blockbuilder/genesisState.go

```diff
@@ -0,0 +1,50 @@
+package blockbuilder
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+var ErrChainIDRequired = errors.New("ChainID is required")
+
+type GenesisConfig struct {
+	ChainID                 string           `json:"chain_id"`
+	WhitelistedSequencerSet []felt.Felt      `json:"whitelisted_sequencer_set"`
+	State                   genesisStateData `json:"state"`
+}
+
+type genesisStateData struct {
+	Classes       map[string]genesisClassData    `json:"classes"`
+	Contracts     map[string]genesisContractData `json:"contracts"`
+	Storage       map[string][]core.StorageDiff  `json:"storage"`
+	FunctionCalls []functionCall                 `json:"function_calls"`
+}
+
+type genesisClassData struct {
+	Path    string `json:"path"`
+	Version int    `json:"version"`
+}
+
+type genesisContractData struct {
+	ClassHash       felt.Felt    `json:"class_hash"`
+	ConstructorArgs *[]felt.Felt `json:"constructor_args"`
+}
+type functionCall struct {
+	ContractAddress    felt.Felt   `json:"contract_address"`
+	EntryPointSelector felt.Felt   `json:"entry_point_selector"`
+	Calldata           []felt.Felt `json:"calldata"`
+}
+
+func (g *GenesisConfig) UnmarshalJSON(data []byte) error {
+	type genesisConfig GenesisConfig
+	if err := json.Unmarshal(data, (*genesisConfig)(g)); err != nil {
+		return err
+	}
+	if g.ChainID == "" {
+		return ErrChainIDRequired
+	}
```

Comment:
Let's not mix unmarshaling and validation. You can use `validator` package for validation. See https://github.com/NethermindEth/juno/blob/acc9ac0f3c3895b5cacf41741df2d8c47e6da294/rpc/transaction.go#L222-L243